### PR TITLE
Clean exit (#98)

### PR DESCRIPTION
**Pull request by sergio-costas**
* Clean exit

This patch manages the INT and TERM signals to cleanly quit the main loop and free all the used resources.

* Clear objects in shutdown event

### CODE REVIEW

**Comment by sergio-costas** [comment]
@robert-ancell I accidentally set as destination branch an incorrect one. The "clean exit" patch should have been merged into main. Any problem with me merging it?

File: src/main.c [hunk 1]
```diff
@@ -150,7 +151,8 @@ static void do_startup(GObject *object, gpointer data) {
 
 static void do_activate(GObject *object, gpointer data) {
   // because, by default, there are no windows, so the application would quit
-  g_application_hold(G_APPLICATION(app));
+  g_application_hold(G_APPLICATION(object));
+  running = TRUE;
```
Comment (3v1n0) [inline src/main.c:155]: Can't you just hold the app by default and avoid the `running` bit?

Comment (sergio-costas) [reply src/main.c:155]: The problem is that pressing Ctrl-C seems to send both SIGINT and SIGTERM signals, so g_application_release() is called twice.

Comment (sergio-costas) [reply src/main.c:155]: (part of the problem is due to the trick used to avoid problems with text-only sessions)

Comment (sergio-costas) [reply src/main.c]: Ok, I found that I can call g_application_quit several times, so that fixes the problem.

**Review by robert-ancell — APPROVED** [review]
LGTM, but perhaps wait for @3v1n0 to finish his review.

**Comment by 3v1n0** [comment]
Thanks for asking, but all good from my side too! :)

**Comment by sergio-costas** [comment]
@robert-ancell @3v1n0 Thanks! Merging.

**Pull request merged**